### ISSUE TITLE
Issues in child workflow affects parent workflow in unexpected way

### ISSUE THREAD

**Issue by WenovateAA** [Azure/logicapps#470]
Built-in invoke workflow action causes many unexpected issues, here is one of them. 
Scenario: Workflow1 invokes Workflow2 using built-in action. Simplest example:

Workflow 1

![image](https://user-images.githubusercontent.com/97520510/153175299-eeddaa4e-ebb0-4d73-9fe1-fba4523b2aae.png)

````json
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {
            "Compose": {
                "inputs": "@body('Invoke_a_workflow_in_this_workflow_app')",
                "runAfter": {
                    "Invoke_a_workflow_in_this_workflow_app": [
                        "Succeeded"
                    ]
                },
                "type": "Compose"
            },
            "Invoke_a_workflow_in_this_workflow_app": {
                "inputs": {
                    "body": {
                        "input": "Request from worflow 1"
                    },
                    "host": {
                        "workflow": {
                            "id": "workflow2"
                        }
                    }
                },
                "runAfter": {},
                "type": "Workflow"
            }
        },
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "triggers": {
            "manual": {
                "inputs": {},
                "kind": "Http",
                "type": "Request"
            }
        }
    },
    "kind": "Stateful"
}
````

First workflow2 works fine. Then write it a way it contains error that allows you to save it. 
For example, use one action's name in next action put to some function. Rename this first action. Designer can handle action rename, but doesn't do this if action put to functions as parameter.
So here is example of wrong Workflow2:

````json
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {
            "Compose_2": {
                "inputs": "@triggerBody()?['input']",
                "runAfter": {},
                "type": "Compose"
            },
            "Response": {
                "inputs": {
                    "body": "@concat(outputs('Compose'))",
                    "statusCode": 200
                },
                "kind": "http",
                "runAfter": {
                    "Compose_2": [
                        "Succeeded"
                    ]
                },
                "type": "Response"
            }
        },
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "triggers": {
            "manual": {
                "inputs": {
                    "schema": {
                        "properties": {
                            "input": {
                                "type": "string"
                            }
                        },
                        "type": "object"
                    }
                },
                "kind": "Http",
                "type": "Request"
            }
        }
    },
    "kind": "Stateful"
}
````
Here `@concat(outputs('Compose'))` referring to non-existing action name `Compose`, but only `Compose_2` exists. 
Upon saving we see error:
![image](https://user-images.githubusercontent.com/97520510/153176464-061e5604-8de4-4471-a11c-62a578f738ec.png)

UPD: another way to break child workflow is to use actions with managed connections I faced. If there are problems with connections (connections.json contains some wrong information), both child and parent workflow stop working too.

Now Workflow1 starts behave in unexpected way. Sometimes you can see these errors trying to trigger it from Designer:

![image](https://user-images.githubusercontent.com/97520510/153176741-d62a38e9-c11d-4122-afe9-e43b064ad31e.png)

If you try to see run history, you see `Unexpected error. Failed to fetch` errors:

![image](https://user-images.githubusercontent.com/97520510/153176857-18b557fa-dae3-4e58-9ad1-600cff2a6ddb.png)

This is because Inputs and Outputs API return 401:

![image](https://user-images.githubusercontent.com/97520510/153177086-9493b51e-60e8-4bea-8148-3cfb43b804bf.png)

To summarize:
1. Errors in child template can unexpectedly cause problems in parent templates
2. There is no obvious way you can understand that they are caused by failed child workflow, errors messages are misleading
3. When trying to invoke parent workflow using HTTP trigger, you get 404 as if workflow just doesn't exist. This adds more confusion.
3. The only way to determine the cause I see is to read Log streaming in Kudu

Overall, this problem makes using invoke workflow actions unsuitable for any production workload
